commit 229bc19b08cfe4336f6529fb9632c2f1b4618c2c
Author: Anthony Mallet <anthony.mallet@laas.fr>
Date:   Sun Nov 20 18:56:33 2011 +0100

    Rewrite filter arg structure for gcc-4.6
    
    Until gcc-4.6, a duplicate member in an unnamed struct/union member of a
    struct/union was not detected. Fix the viam_filterarg_obj union to avoid a
    duplicate 'vkind' member. No functional change.

diff --git a/include/viam/viam_acq_driver.h b/include/viam/viam_acq_driver.h
index ce3997b..fd3adf9 100644
--- include/viam/viam_acq_driver.h
+++ include/viam/viam_acq_driver.h
@@ -49,12 +49,11 @@ typedef union viam_filterargs_obj {
   };					/**< crop, scale and the like */
   struct {
     viam_valuekind_t vkind;
-    double blue, red;
-  };					/**< white balance */
-  struct {
-    viam_valuekind_t vkind;
-    double value;
-  };					/**< single valued filters */
+    union {
+      double value;			/**< single valued filters */
+      double blue, red;			/**< white balance */
+    };
+  };
   viam_filterformat_t format;		/**< for format filters */
 } *viam_filterargs_obj;
 
